metricbeat/module/kubernetes/util/kubernetes.go

```diff
@@ -214,27 +168,24 @@ func NewContainerMetadataEnricher(
 	base mb.BaseMetricSet,
 	nodeScope bool) Enricher {
 
-	watcher, err := GetNamedWatcher("container_metadata_enricher", base, &kubernetes.Pod{}, nodeScope)
-	if err != nil {
-		logp.Err("Error initializing Kubernetes metadata enricher: %s", err)
-		return &nilEnricher{}
-	}
-
-	if watcher == nil {
+	config := validatedConfig(base)
+	if config == nil {
 		logp.Info("Kubernetes metricset enriching is disabled")
 		return &nilEnricher{}
 	}
 
-	metaConfig := metadata.Config{}
-	if err := base.Module().UnpackConfig(&metaConfig); err != nil {
+	watcher, nodeWatcher, namespaceWatcher := getPodMetadataWatchers(config, &kubernetes.Pod{}, nodeScope)
```

Comment:
This one actually could cover the TODOs like at https://github.com/elastic/beats/pull/29133/files#diff-795a7b218ad6ff27d43bc453ecce7984c3d816537b1afc9cfc9000d5d574119eR212?

Reply:
I've created a dedicated issue for this: https://github.com/elastic/beats/issues/29200 and assigned to me